### PR TITLE
CmdStan 2.36 no longer requires fixed_param hacks

### PR DESCRIPTION
**Pull request by WardBrian**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and open-source license: see below

#### Summary

CmdStan 2.36 allows the HMC samplers with 0 parameters

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):
Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by WardBrian** [comment]
> FAILED test/test_sample.py::test_sample_no_params - TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''...

weird, that’s the error this branch gets if you run it with an older cmdstan. I don’t get that with the RC

I didn’t try from_csv yet, but I’m a bit surprised at that error, since the changes here still do create a (empty) matrix

**Comment by WardBrian** [comment]
After the latest round of Stan PRs, this branch is now passing:

https://github.com/stan-dev/cmdstanpy/actions/runs/12091362509

there might be more fixes needed on our end for things like the from_csv method, but I am guessing that RC2 will be more or less done

**Comment by WardBrian** [comment]
Passed under RC2: https://github.com/stan-dev/cmdstanpy/actions/runs/12141600359

File: cmdstanpy/model.py [hunk 1]
```diff
@@ -205,9 +205,9 @@ def __init__(
                 self._compiler_options.add_include_path(path)
 
             # try to detect models w/out parameters, needed for sampler
-            if not cmdstan_version_before(
-                2, 27
-            ):  # unknown end of version range
+            if not cmdstan_version_before(2, 27) and cmdstan_version_before(
+                2, 36
+            ):
```
Comment (mitzimorris) [inline cmdstanpy/model.py:210]: I would add parans around the not just to make things super easy to grok:

`if (not cmdstan_version_before(2, 27)) and cmdstan_version_befor(2,36):`

File: test/test_sample.py [hunk 2]
```diff
@@ -1621,33 +1637,18 @@ def test_validate_sample_sig_figs(stanfile='bernoulli.stan'):
...
-        sig_figs=17,
-        method_args=sampler_args,
+    # construct CmdStanMCMC from logistic model output
+    fit = from_csv(
+        [
+            os.path.join(DATAFILES_PATH, 'logistic_output_1.csv'),
+            os.path.join(DATAFILES_PATH, 'logistic_output_2.csv'),
+            os.path.join(DATAFILES_PATH, 'logistic_output_3.csv'),
+            os.path.join(DATAFILES_PATH, 'logistic_output_4.csv'),
+        ]
```
Comment (mitzimorris) [inline test/test_sample.py:1647]: nice!  much cleaner

**Review by mitzimorris — APPROVED** [review]
LGTM!  nice to be rid of that hack.

**Pull request merged**